src/ParseObject.js

```diff
@@ -316,11 +342,16 @@ export default class ParseObject {
 
   _migrateId(serverId: string) {
     if (this._localId && serverId) {
-      var oldState = ObjectState.removeState(this.className, this._getStateIdentifier());
-      this.id = serverId;
-      delete this._localId;
-      if (oldState) {
-        ObjectState.initializeState(this.className, this._getStateIdentifier(), oldState);
+      if (singleInstance) {
+        var oldState = SingleInstanceState.removeState(this.className, this._getStateIdentifier());
+        this.id = serverId;
+        delete this._localId;
+        if (oldState) {
+          SingleInstanceState.initializeState(this.className, this._getStateIdentifier(), oldState);
```

Comment:
Even here, you can pass `this` to `UniqueInstanceState` and just do nothing in `removeState` and `initializeState`.